app/models/project.rb

```diff
@@ -27,4 +29,10 @@ class Project < ApplicationRecord
   def generate_first_page
     pages.build(title: title, source: '')
   end
+
+  private
+
+  def downcase_slug
+    self.slug &&= slug.downcase
```

Comment:
На мой взгляд так нагляднее

```suggestion
    self.slug = slug.presence&.downcase
```

Reply:
А мне как-то первое привычней. Так что оставлю как у меня.